src/derive.rs

```diff
@@ -283,15 +283,15 @@ pub trait Subcommand: FromArgMatches + Sized {
 /// ```
 pub trait ArgEnum: Sized {
     /// All possible argument choices, in display order.
-    const VARIANTS: &'static [&'static str];
+    fn variants() -> Vec<ArgValue<'static>>;
 
     /// Parse an argument into `Self`.
     fn from_str(input: &str, case_insensitive: bool) -> Result<Self, String>;
 
     /// The canonical argument value.
     ///
     /// The value is `None` for skipped variants.
-    fn as_arg(&self) -> Option<&'static str>;
+    fn as_arg(&self) -> Option<ArgValue<'static>>;
```

Comment:
If we are changing the return type, we should make sure the name matches.  I chose `as_arg` to say that this is what the argument would look like.

I think the name should be consistent with what we rename `variants` to, whether
- `arg_value`
- `value`